aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs

```diff
@@ -155,6 +157,36 @@ impl<'r> TResourceView for ExecutorViewWithChangeSet<'r> {
     }
 }
 
+impl<'r> TResourceGroupView for ExecutorViewWithChangeSet<'r> {
+    type Key = StateKey;
+    type Layout = MoveTypeLayout;
+    type Tag = StructTag;
+
+    fn resource_group_size(&self, _state_key: &Self::Key) -> anyhow::Result<u64> {
+        // In respawned session, gas is irrelevant, so we return 0 (GroupSizeKind::None).
+        Ok(0)
+    }
+
+    fn get_resource_from_group(
+        &self,
+        state_key: &Self::Key,
+        resource_tag: &Self::Tag,
+        maybe_layout: Option<&Self::Layout>,
+    ) -> anyhow::Result<Option<Bytes>> {
+        self.change_set
+            .resource_group_write_set()
+            .get(state_key)
+            .and_then(|g| g.inner_ops.get(resource_tag))
+            .map(|op| op.extract_raw_bytes())
+            .ok_or(anyhow::Error::msg("Must be ignored immediately after"))
```

Comment:
this error message doesn't make much sense to me, also the whole block logic is not very easy to follow. I'd just write it like this
```
        if let Some(write_op) = self
            .change_set
            .resource_group_write_set()
            .get(state_key)
            .and_then(|g| g.inner_ops.get(resource_tag))
        {
            Ok(write_op.extract_raw_bytes())
        } else {
            self.base
                .get_resource_from_group(state_key, resource_tag, maybe_layout)
        }
```

Reply:
nice